keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/SamplingAlgorithm.java

```diff
@@ -8,20 +8,23 @@
 public interface SamplingAlgorithm {
 
     /**
-     * Same effect as a sample but the result isn't saved or returned.
+     * Move forward the state of the Sampling Algorithm by a single step.
      */
     void step();
 
     /**
-     * Takes a sample with the algorithm and saves it in the supplied map
+     * Takes a sample with the algorithm and saves it in the supplied map.  Repeated calls to this function will return
```

Comment:
> Repeated calls to this function will return the same values without an intermediary call to `step()`

I don't follow. This method doesn't explicitly return anything? And repeated calls will update the sample list for each vertex.

Reply:
Good catch - I was documenting the wrong fix.  Made changes.